File: ip-messaging/rest/channels/list-channels/list-channels.5.x.cs
// Download the twilio-csharp library from twilio.com/docs/libraries/csharp
using System;
using Twilio;
using Twilio.Rest.Chat.V1.Service;

class Example
{
    static void Main (string[] args)
    {
        // Find your Account SID and Auth Token at twilio.com/console
        const string accountSid = "accountSid";
        const string authToken = "authToken";
        const string serviceSid = "serviceSid";

        TwilioClient.Init(accountSid, authToken);

        //Retrieve the list of channels
        var channels = ChannelResource.Read(serviceSid);
        foreach (var channel in channels)
        {
            Console.WriteLine(channel.FriendlyName);
        }
    }
}

File: ip-messaging/rest/channels/list-channels/list-channels.5.x.rb
require 'twilio-ruby'

# Get your Account SID and Auth Token from twilio.com/console
account_sid = 'ACCOUNT_SID'
auth_token = 'AUTH_TOKEN'
service_sid = 'SERVICE_SID'
@client = Twilio::REST::Client.new(account_sid, auth_token)

# List the channels
service = @client.chat.v2.services(service_sid)
channels = service.channels.each do |c|
    puts "#{c.sid} \"#{c.unique_name}\""
end
